src/sheet/components/SheetDataEditorCell.tsx

```diff
@@ -52,7 +53,9 @@ export default function SheetDataEditorCell({
     columnDefinition.type === 'enum'
       ? (columnDefinition.typeArguments.values.find((e) => e.value === value)
           ?.label ?? value)
-      : value;
+      : columnDefinition.type === 'reference'
```

Comment:
Probably this whole thing (extracting non empty value) could be extracted to a util?